packages/client/lib/client/authx/credentials-provider.ts

```diff
@@ -0,0 +1,103 @@
+import {Disposable} from './types';
```

Comment:
Would it be worth using TS 5.2 "native" `Disposable` interface (https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-2.html#using-declarations-and-explicit-resource-management)

Reply:
@dhensby makes total sense, i will use the built-in Disposable, thanks for the tip!

Reply:
`Symbol.dispose` is not meant to be called directly, but to be used together with `using` and automatically called when the "using variable" goes out of scope (see the example from the typescript article). I don't think it fits here..

see https://github.com/tc39/proposal-explicit-resource-management for more details..